### PR TITLE
Feature: search decks

### LINKED ISSUE #193: Flashcards: add decks to search menu
Modify search menu to be able to search for public decks by name / tags. Make filter chips row scrollable to make room

### PR DESCRIPTION
**Pull request by robertaleG**
# Description
Integrate decks with search menu: add a new filter chip for decks and deck items which show relevant info and redirect to DECK_MENU screen (which is now empty but will be implemented in PR #184).

### Changes:  
- **Deck** : added fields `lastModified: Timestamp` for UX and `visibility: Visibility` to decks, for the search bar to only be able to search through public decks, 
- **SearchMenu** : now displays all elements of the selected filter chip when the query is empty

Fixes #193 

# How Has This Been Tested?

### UI Tests
I adapted SearchTest to test the deck filter chip, deck items and redirection towards deck menu page.

### End to end Tests
Minor modifications to remove warnings and make code compile.
  
# Dependencies

None added

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged in downstream modules, no conflicts with main

### CODE REVIEW

File: app/src/main/java/com/github/onlynotesswent/ui/search/Search.kt [hunk 1]
```diff
@@ -278,29 +338,87 @@ fun SearchScreen(
...
 
+/**
+ * Displays a deck item in the search screen.
+ *
+ * @param deck The deck to display.
+ * @param author The author of the deck.
+ * @param onClick The action to perform when the deck is clicked.
+ */
+@Composable
+fun DeckSearchItem(deck: Deck, author: String, onClick: () -> Unit) {
```
Comment (Roshan-y) [inline app/src/main/java/com/github/onlynotesswent/ui/search/Search.kt:349]: Not needed for this pr. But making this a common deck card like for note would have allowed us to reuse this for the overview of the deck.

**Review by Roshan-y — CHANGES_REQUESTED** [review]
The deck card ui looks good in the search screen. The tests have been correctly updated to test the search and the new deck model. Was wondering why we are passing the deck id in the navigation. The deck card could have been modularised, but not necessary to do now.

File: app/src/main/java/com/github/onlynotesswent/ui/search/Search.kt (hunk 1 above)
Comment (robertaleG) [reply app/src/main/java/com/github/onlynotesswent/ui/search/Search.kt:349]: yes I can do that

File: app/src/main/java/com/github/onlynotesswent/model/flashcard/deck/DeckViewModel.kt [hunk 2]
```diff
@@ -53,6 +57,16 @@ class DeckViewModel(private val repository: DeckRepository) : ViewModel() {
     return repository.getNewUid()
   }
 
+  /** Retrieves all public decks. */
+  fun getPublicDecks(onSuccess: (List<Deck>) -> Unit = {}, onFailure: (Exception) -> Unit = {}) {
```
[the remark this answers is not in the document]
Comment (robertaleG) [reply app/src/main/java/com/github/onlynotesswent/model/flashcard/deck/DeckViewModel.kt:61]: thanks

[the remark this answers is not in the document]
Comment (robertaleG) [reply app/src/main/java/com/github/onlynotesswent/model/flashcard/deck/DeckViewModel.kt:61]: and I will add the callbacks parameters to the doc of the function

File: app/src/main/java/com/github/onlynotesswent/ui/navigation/NavigationActions.kt [hunk 3]
```diff
@@ -33,6 +33,7 @@ object Screen {
   const val EDIT_PROFILE = "Edit Profile Screen"
   const val NOTIFICATIONS = "Notifications Screen"
   const val FOLDER_CONTENTS = "Folder Contents Screen/{folderId}"
+  const val DECK_MENU = "Deck Menu Screen/{deckId}"
```
[the remark this answers is not in the document]
Comment (robertaleG) [reply app/src/main/java/com/github/onlynotesswent/ui/navigation/NavigationActions.kt:36]: actually I'll do that in the deck menu pr instead, not search

**Review by Roshan-y — APPROVED** [review]

**Pull request merged**